its-a-smol-world/src/app.py

```diff
@@ -0,0 +1,46 @@
+import time
+import itertools
+import threading
+import sys
+
+from smol_mind import SmolMind, load_functions
+MODEL_NAME = "HuggingFaceTB/SmolLM-1.7B-Instruct"
+
+def spinner(stop_event):
+    spinner = itertools.cycle(['-', '/', '|', '\\'])
+    while not stop_event.is_set():
+        sys.stdout.write(next(spinner))
+        sys.stdout.flush()
+        sys.stdout.write('\b')
+        time.sleep(0.1)
+
+def main():
```

Comment:
Overkill for this example, but for more complex demos we can use [Textual](https://github.com/Textualize/textuall) in the future.